extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessor.java

```diff
@@ -66,37 +60,25 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.function.Supplier;
 
 public class WindowOperatorQueryFrameProcessor implements FrameProcessor<Object>
 {
   private static final Logger log = new Logger(WindowOperatorQueryFrameProcessor.class);
-  private final WindowOperatorQuery query;
 
   private final List<OperatorFactory> operatorFactoryList;
-  private final List<String> partitionColumnNames;
-  private final ObjectMapper jsonMapper;
-  private final ArrayList<RowsAndColumns> frameRowsAndCols;
   private final ArrayList<RowsAndColumns> resultRowAndCols;
+  private final RowsAndColumnsBuilder frameRowsAndColsBuilder;
   private final ReadableFrameChannel inputChannel;
   private final WritableFrameChannel outputChannel;
   private final FrameWriterFactory frameWriterFactory;
   private final FrameReader frameReader;
   private final int maxRowsMaterialized;
-  private Cursor frameCursor = null;
-  private Supplier<ResultRow> rowSupplierFromFrameCursor;
-  private ResultRow outputRow = null;
   private FrameWriter frameWriter = null;
 
   private final VirtualColumns frameWriterVirtualColumns;
   private final SettableLongVirtualColumn partitionBoostVirtualColumn;
 
-  // List of type strategies to compare the partition columns across rows.
-  // Type strategies are pushed in the same order as column types in frameReader.signature()
-  private final NullableTypeStrategy[] typeStrategies;
-
-  private final ArrayList<ResultRow> rowsToProcess;
-  private int lastPartitionIndex = -1;
+  private Operator op = null;
```

Comment:
can this field have a more verbose name?

Reply:
Renamed to `operator`. Any other naming suggestion? 😅